File: Games/GameCollection/src/common/updates/UpdateMainMenu.java
package common.updates;

import common.GameContainer;
import common.MainMenu;
import common.util.Logger;

public class UpdateMainMenu {

    public static void update( MainMenu menu, Object arg, GameContainer gameContainer ) {
        if ( arg instanceof String ) {
            String message = (String) arg;
            if ( gameContainer.containsGame( message ) ) {
                gameContainer.setGameShown( message );
            }
            else if ( message.equals( UpdateCodes.MainMenu.shutdown ) ) {
                gameContainer.stopContainer();
            }
        }
        else {
            Logger.log( Updater.parsingErrorCode + " update(MainMenu, Object)" );
        }
    }

}

File: Games/GameCollection/src/main/common/updates/UpdateMainMenu.kt
package common.updates

import main.common.GameContainer
import common.MainMenu
import common.logger.Logger

object UpdateMainMenu {
    fun update(menu: MainMenu?,
               arg: Any?,
               gameContainer: GameContainer) {
        if (arg is String) {
            if (gameContainer.containsGame(arg)) {
                gameContainer.setGameShown(arg)
            } else if (arg == UpdateCodes.MainMenu.shutdown) {
                gameContainer.stopContainer()
            }
        } else {
            Logger.log(Updater.parsingErrorCode + " update(MainMenu, Object)")
        }
    }
}
